### DISCUSSION TITLE
UDM SE issue: igmpproxy[13621]: MC-Router API already in use; Errno(98): Address already in use

### DISCUSSION THREAD

**Discussion by RDGK74** [fabianishere/udm-iptv#270] [Q&A]
hello,

This week my UDM SE arrived and with the instructies on 'https://github.com/fabianishere/udm-iptv#global-design' I tried to make IPTV work with the ISP KPN (Netherlands). I have tried many things already but the issue what I;m facing now is "igmpproxy[13621]: MC-Router API already in use; Errno(98): Address already in use".
I have removed and reinstalled (see log below). Can anyone point me in the right direction? Thx

Richard

Last login: Fri Aug 18 13:56:15 2023 from 192.168.1.154
root@Dream-Machine-Special-Edition:~# sh -c "$(curl https://raw.githubuserconten                                                                                                                                                             t.com/fabianishere/udm-iptv/master/install.sh -sSf)"
Downloading packages...
Installing packages...
Hit:1 http://security.debian.org/debian-security bullseye-security InRelease
Hit:2 http://deb.debian.org/debian bullseye InRelease
Hit:3 http://deb.debian.org/debian bullseye-updates InRelease
Hit:4 http://deb.debian.org/debian bullseye-backports InRelease
Hit:5 https://apt.artifacts.ui.com bullseye InRelease
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  dialog
0 upgraded, 1 newly installed, 0 to remove and 19 not upgraded.
Need to get 0 B/276 kB of archives.
After this operation, 1225 kB of additional disk space will be used.
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based fronte                                                                                                                                                             nd cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> li                                                                                                                                                             ne 1.)
debconf: falling back to frontend: Readline
Selecting previously unselected package dialog.
(Reading database ... 111905 files and directories currently installed.)
Preparing to unpack .../dialog_1.3-20201126-1_arm64.deb ...
Unpacking dialog (1.3-20201126-1) ...
Setting up dialog (1.3-20201126-1) ...
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  igmpproxy udm-iptv
0 upgraded, 2 newly installed, 0 to remove and 19 not upgraded.
Need to get 0 B/38.3 kB of archives.
After this operation, 167 kB of additional disk space will be used.
Get:1 /tmp/tmp.yLG9uGk7N9/udm-iptv.deb udm-iptv all 3.0.2 [16.3 kB]
Preconfiguring packages ...
/tmp/udm-iptv.config.DOxfog: 142: .: cannot open /usr/lib/udm-iptv/profiles/kpn.                                                                                                                                                             profile: No such file
udm-iptv failed to preconfigure, with exit status 2
Selecting previously unselected package igmpproxy.
(Reading database ... 112068 files and directories currently installed.)
Preparing to unpack .../igmpproxy_0.3-1_arm64.deb ...
Unpacking igmpproxy (0.3-1) ...
Selecting previously unselected package udm-iptv.
Preparing to unpack .../tmp.yLG9uGk7N9/udm-iptv.deb ...
Unpacking udm-iptv (3.0.2) ...
Setting up igmpproxy (0.3-1) ...
Setting up udm-iptv (3.0.2) ...
W: Repository is broken: udm-iptv:arm64 (= 3.0.2) has no Size information
Installation successful... You can find your configuration at /etc/udm-iptv.conf                                                                                                                                                             .

Use the following command to reconfigure the script:

         udm-iptv reconfigure
root@Dream-Machine-Special-Edition:~# udm-iptv diagnose
Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: br4
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: true
=== IP Link and Route ===
=== Service Logs ===
Aug 18 15:25:35 Dream-Machine-Special-Edition udm-iptvd[140580]: Device iptv alr                                                                                                                                                             eady exists.. Deleting device
Aug 18 15:25:35 Dream-Machine-Special-Edition udm-iptvd[140580]: Obtaining IP ad                                                                                                                                                             dress for VLAN interface
Aug 18 15:25:35 Dream-Machine-Special-Edition udm-iptvd[140602]: udhcpc: started                                                                                                                                                             , v1.30.1
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140602]: udhcpc: sending                                                                                                                                                              discover
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140602]: udhcpc: sending                                                                                                                                                              select for 10.142.133.198
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140602]: udhcpc: lease o                                                                                                                                                             f 10.142.133.198 obtained, lease time 3600
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140580]: NATing IPTV net                                                                                                                                                             work ranges (if necessary)
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140580]: Creating static                                                                                                                                                              routes (if necessary)
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140580]: Setting up IGMP                                                                                                                                                              Proxy
Aug 18 15:25:36 Dream-Machine-Special-Edition udm-iptvd[140580]: Using igmpproxy                                                                                                                                                             ...
root@Dream-Machine-Special-Edition:~# systemctl status igmpproxy
● igmpproxy.service - LSB: IGMP multicast routing daemon
     Loaded: loaded (/etc/init.d/igmpproxy; generated)
     Active: inactive (dead)
       Docs: man:systemd-sysv-generator(8)

Aug 18 14:11:34 Dream-Machine-Special-Edition systemd[1]: Started LSB: IGMP multicast routing daemon.
Aug 18 14:33:19 Dream-Machine-Special-Edition systemd[1]: Stopping LSB: IGMP multicast routing daemon...
Aug 18 14:33:19 Dream-Machine-Special-Edition systemd[1]: igmpproxy.service: Succeeded.
Aug 18 14:33:19 Dream-Machine-Special-Edition systemd[1]: Stopped LSB: IGMP multicast routing daemon.
Aug 18 14:33:38 Dream-Machine-Special-Edition systemd[1]: Starting LSB: IGMP multicast routing daemon...
Aug 18 14:33:38 Dream-Machine-Special-Edition systemd[1]: Started LSB: IGMP multicast routing daemon.
Aug 18 14:33:38 Dream-Machine-Special-Edition igmpproxy[13621]: MC-Router API already in use; Errno(98): Address already in use
Aug 18 14:57:43 Dream-Machine-Special-Edition systemd[1]: Stopping LSB: IGMP multicast routing daemon...
Aug 18 14:57:43 Dream-Machine-Special-Edition systemd[1]: igmpproxy.service: Succeeded.
Aug 18 14:57:43 Dream-Machine-Special-Edition systemd[1]: Stopped LSB: IGMP multicast routing daemon.
root@Dream-Machine-Special-Edition:~#

**Comment by fabianishere** [comment]
udm-iptv doesn’t use igmpproxy.service, so you shouldn’t try to enable/start it. The error you are receiving indicates that two igmpproxy instances are trying to run concurrently.

According to the logs, it seems udm-iptv managed to get a DHCP lease and successfully started igmpproxy. I suspect there’s an issue with the way you have connected the STB to your network.

**Comment by RDGK74** [comment]
Dear Fabian,
Thank you for your analysis and response. Before I had a fritzbox form my ISP (KPN Netherlands) placed before my UDM SE, and with that configuration it worked. My assumption was (and is)  that de STB should be connected correctly. 
The settings of VLAN4 are:

![image](https://github.com/fabianishere/udm-iptv/assets/142602337/6dbe9756-d89e-4aab-8cbe-aea3ce0f9353)

I hope you can look again, and give another suggestion.

For now thx.

Richard

**Reply by fabianishere** [reply]
Did you set the port profile on the port that connects the STB?

**Comment by RDGK74** [comment]
Hi Fabian,

Yes both ports are set for VLAN4 (see screenprint).

![image](https://github.com/fabianishere/udm-iptv/assets/142602337/2c03f4f2-da25-4d78-b580-97c212a7bce8)

I really appreciate your help and hope you have another suggestion, where I could look.

Richard

**Reply by M-Arends** [reply]
Same issue here.UDMP Pro (not SE)  I removed multicast DNS  on the iptv vlan and a apt-get update. And now it works . 

p.s. I have a IGMP proxy option in the Global network section that is disabled in my case(see below).

![image](https://github.com/fabianishere/udm-iptv/assets/99212621/acc58a37-fbbc-441a-a9e1-5d463ab22e4a)

**Reply by RDGK74** [reply]
Dear M-Arends, this was the last hint I needed, it works now! I want to thank you and Fabian for getting me there. There are some happy people in my house right now. THX!! 
Richard
